java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture4/tasks/_4/Number4.java

```diff
@@ -0,0 +1,131 @@
+package com.zoolatech.lecture4.tasks._4;
+
+/*
+Create an enum class that represents the following weight units: gram, kilogram, ounce, pound, ton.
+The class should provide methods to get an abbreviation for a unit (g, kg, oz, lb, t),
+convert a value from a source unit to a current one (method should accept a value and a source unit),
+and method that allows to get the enum value by a string representation, that can be passed in any case
+(e.g., for strings “gram”, “Gram”, “GRAM” the method should return an enum value for the gram unit).
+*/
+
+public class Number4 {
+    public static void main(String[] args) {
+        Weight weight;
+        System.out.println(Weight.abbreviation(Weight.KILOGRAM));
+        System.out.println(Weight.enumValueByAString("lb"));
+        System.out.println(Weight.convertValue(1, Weight.GRAM, Weight.KILOGRAM));
+    }
+}
+
+enum Weight {
+    GRAM,
+    KILOGRAM,
+    OUNCE,
+    POUND,
+    TON;
+
+    static String abbreviation(Weight weight) {
+        return switch (weight) {
+            case GRAM -> "gr";
+            case KILOGRAM -> "kg";
+            case OUNCE -> "oz";
+            case POUND -> "lb";
+            case TON -> "t";
+        };
+    }
+
+    static String convertValue(float mass, Weight sourceUnit, Weight convertTo) {
+        //MARK: GRAM
+        if (sourceUnit == GRAM && convertTo == KILOGRAM) {
```

Comment:
you could simplify your code if you do smth like this:
```
if(sourceUnit == GRAM) {
    if(convertTo == KILOGRAM){
    
    } else if (convertTo ...)
} else if (sourceUnit == KILOGRAM) {
...
}
```
you could even use switch statements instead of if-else blocks